Microsoft.AVS.VVOLS/Microsoft.AVS.VVOLS.psm1

```diff
@@ -0,0 +1,370 @@
+    {
+        "SchemaVersion": "1.0.0",
+        "Vendor": "Pure Storage",
+        "PolicyName": "",
+        "PolicyDescription": "",
+        "SpbmRules": {
+            "com.purestorage.storage.policy.FlashArrayGroup":["fa1","fa2"],
+            "com.purestorage.storage.replication.LocalSnapshotPolicyCapable": true,
+            "com.purestorage.storage.replication.LocalSnapshotInterval":"00:00:00"
+            ...
+        }
+    }
+
+    .EXAMPLE
+     New-VvolStoragePolicy -PolicyConfigJsonString $policyConfigJsonString
+
+    .INPUTS
+     vCenter storage policy in Json string
+
+    .OUTPUTS
+     None.
+#>
+function New-VvolStoragePolicy {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param(
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Storage policy in Json string')]
+        [ValidateNotNull()]
+        [String]$PolicyConfigJsonString
+    )
+    # Convert JSON to PowerShell object
+    $PolicyConfig = ConvertFrom-Json $PolicyConfigJsonString
+    # Extract values from the object
+    $Vendor = $PolicyConfig.Vendor
+    $PolicyName = $PolicyConfig.PolicyName
+    $PolicyDescription = $PolicyConfig.PolicyDescription
+    $SchemaVersion = $PolicyConfig.SchemaVersion
+
+    if ($SchemaVersion -gt "1.0.0") {
+        throw "Unable to create a storage policy. Schema version $SchemaVersion is not supported."
+    }
+
+    # Create SPBM rules dictionary
+    $rules = @()
+    $PolicyConfig.SpbmRules.PSObject.Properties | ForEach-Object {
+        $RuleName = $_.Name
+        $RuleValue = $_.Value
+        if ($Vendor -eq "Pure Storage") {
+            $ValueType = (Get-SpbmCapability -Name $RuleName).ValueType.Name
+            if ($RuleValue -is [int64] -and $ValueType -eq "int32") {
+                $RuleValue = [int]$RuleValue
+            }
+            if ($ValueType -eq "TimeSpan") {
+                $RuleValue = [TimeSpan]::Parse($RuleValue)
+            }
+        }
+        $rules += New-SpbmRule -Capability (Get-SpbmCapability -Name $RuleName) -Value $RuleValue
+    }
+
+    Write-Host "Creating policy $PolicyName for vendor $Vendor..."
+    $Ruleset = New-SpbmRuleSet -AllOfRules $rules
+    New-SpbmStoragePolicy -Name $PolicyName -Description $PolicyDescription -AnyOfRuleSets $Ruleset -ErrorAction Stop | Out-Null
+}
+
+<#
+    .SYNOPSIS
+    Removes a vVol Storage Policy
+
+    .PARAMETER PolicyName
+    Name of the storage policy to be removed
+
+    .EXAMPLE
+    Remove-VvolStoragePolicy -PolicyName "myPolicy"
+
+    .INPUTS
+    vCenter storage policy name
+
+    .OUTPUTS
+    None.
+#>
+function Remove-VvolStoragePolicy {
+    [CmdletBinding()]
+    [AVSAttribute(10, UpdatesSDDC = $false)]
+    Param(
+        [Parameter(
+            Mandatory = $true,
+            HelpMessage = 'Storage policy name')]
+        [ValidateNotNull()]
+        [String]$PolicyName
+    )
+
+    $Policy = Get-SpbmStoragePolicy -Name $PolicyName -ErrorAction Ignore
```

Comment:
Ditto.

Reply:
Same we can not check which objects are using the policy at the time it is created